menas/ui/service/DialogFactory.js

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class EntityDialogFactory {
+
+  constructor(oController, fnLoad, fragmentName, dialogId) {
+    this._oController = oController;
+    let oView = oController.getView();
+
+    fnLoad({
+      id: oView.getId(),
+      name: fragmentName,
+      controller: oController
+    }).then(function (oDialog) {
+      oView.addDependent(oDialog);
+    });
+
+    this._oDialog = oController.byId(dialogId);
+  }
+
+  get oController() {
+    return this._oController;
+  }
+
+  get oDialog() {
+    return this._oDialog;
+  }
+
+}
+
+class SchemaDialogFactory extends EntityDialogFactory {
+
+  constructor(oController, fnLoad) {
+    super(oController, fnLoad, "components.schema.addSchema", "addSchemaDialog")
```

Comment:
Not final, but initially we went with always add `;` at the end, so as to not have to worry about the rules for when they are inferred and when not. If we started this fresh, I'd probably avoid them all together. The thing is, if you know the rules for when they are necessary and when they can be safely left out, why bother adding them at all? It only makes sense to add them out of fear that you'll miss a case where the semicolon cannot be inferred.

Reply:
> ... The thing is, if you know the rules for when they are necessary and when they can be safely left out, why bother adding them at all? ...

It might because it is then easier to do changes, not being concerned about lines around your code. And probably for new people, it is also more "secure"